src/Security/UserProvider/CustomerObjectUserProvider.php

```diff
@@ -74,26 +76,32 @@ public function loadUserByUsername($username)
 
     /**
      * @inheritdoc
+     *
+     * @return UserInterface
      */
-    public function refreshUser(UserInterface $user)
+    public function refreshUser(UserInterface $user)//: UserInterface
     {
         $class = $this->customerProvider->getCustomerClassName();
         if (!$user instanceof $class || !$user instanceof AbstractObject) {
             throw new UnsupportedUserException();
         }
 
-        return $this->customerProvider->getById($user->getId(), true);
+        $customer = $this->customerProvider->getById($user->getId(), true);
+
+        if (!$customer instanceof UserInterface) {
+            throw new UserNotFoundException(sprintf('Customer "%s" was not found', $user->getId()));
+        }
+
+        return $customer;
     }
 
     /**
      * @inheritdoc
+     *
+     * @return bool
      */
-    public function supportsClass($class)
+    public function supportsClass($class)//: bool
```

Comment:
```suggestion
    public function supportsClass($class)
```
Return type hint not added on Symfony 6